File: modkl.cpp
#include <iostream>
using namespace std;
#pragma GCC optimize("trapv")
#define int long long
int find_mth_number(long long N, long long K, long long M) {
    long long count_per_segment = N / K;

    for (long long remainder = 0; remainder < K; ++remainder) {
        if (M <= count_per_segment) {
            return remainder + K * (M - 1);
        }
        M -= count_per_segment;
    }

    return -1;
}

signed main() {
    long long N, K, M;
    cin >> N >> K >> M;
    cout << find_mth_number(N, K, M) << endl;
    return 0;
}

File: modkl.py
def find_mth_number(N, K, M):
    count_per_segment = N // K

    segment_index = (M - 1) // count_per_segment

    if segment_index >= K:
        return -1
    if segment_index == 0:
        return M * K
    return segment_index + K * ((M - 1) % count_per_segment)

N, K, M = map(int, input().split())
print(find_mth_number(N, K, M))